src/imitation/algorithms/adversarial.py

```diff
@@ -259,11 +269,17 @@ def train_gen(self, total_timesteps: Optional[int] = None, callback=None):
       callback: Callback argument to the Stable Baselines `RLModel.learn()`
         method.
     """
+    if self.error_on_unexpected_policy_change:
+      if self.gen_policy.env is not self.venv_train_norm_buffering:
+        raise ValueError("Unexpected change to self.gen_policy.env")
+      if self.venv_train_norm_buffering.n_resets > 1:
```

Comment:
This check doesn't seem to quite match the description in the docstring. In particular, I could still reset (maybe indirectly, by calling a `rollout` method) on `self.venv_train` or `self.venv_train_norm` and I don't think `self.venv_train_norm_buffering` would observe this.

In general not sure it's tractable to detect this. If we're really paranoid user should pass in a thunk to construct a `VecEnv`, and then we construct it internally to `AdversarialTrainer` and do not expose it publicly.

Reply:
>  I could still reset (maybe indirectly, by calling a rollout method) on self.venv_train or self.venv_train_norm and I don't think self.venv_train_norm_buffering would observe this.

That's a really good point